discojs/src/processing/text.ts

```diff
@@ -1,47 +1,56 @@
-import { List, Repeat } from "immutable";
+import { List } from "immutable";
 import { PreTrainedTokenizer } from "@xenova/transformers";
+import type { Text, TokenizedText } from '../index.js'
 
 function isArrayOfNumber(raw: unknown): raw is number[] {
   return Array.isArray(raw) && raw.every((e) => typeof e === "number");
 }
 
-type Token = number;
+interface TokenizingConfig {
```

Comment:
can we have a smaller config? many of theses fields are arcane to me.

Reply:
I'm removing the non-essential ones but LLM generation is inherently complex (see the [HuggingFace docs](https://huggingface.co/docs/transformers/en/generation_strategies) for an overview of generation strategies)